src/zjs_util.h

```diff
@@ -39,12 +53,15 @@ void *zjs_malloc_with_retry(size_t size);
 #define zjs_malloc(sz)                                                      \
     ({                                                                      \
         void *zjs_ptr = zjs_malloc_with_retry(sz);                          \
-        ZJS_PRINT("%s:%d: allocating %lu bytes (%p)\n", __func__, __LINE__, \
+        ZJS_PRINT("%s:%d: allocating %u bytes (%p)\n", __func__, __LINE__,  \
                   (u32_t)sz, zjs_ptr);                                      \
+        char *func_copy = malloc(strlen(__func__) + 1);                     \
+        strcpy(func_copy, __func__);                                        \
+        zjs_push_mem_stat(zjs_ptr, __FILE__, func_copy, __LINE__);          \
```

Comment:
I don't think you need to copy the function name like this, which is nice - it will save RAM while you're using this feature. When you use __func__, it will cause the compiler to store the function name as a static string in the program's data segment. So you can pass it around as a const char * and it will always be there; then you don't need to free it.

So basically here remove func_copy and replace it with __func__ in the push call. Then make sure to stop freeing it above, just NULL it out instead. And you probably need to make it const char *func in the header file.